loader/unknown_function_handling.c

```diff
@@ -0,0 +1,322 @@
+            while (ldev) {
+                gdpa_value = ldev->loader_dispatch.core_dispatch.GetDeviceProcAddr(ldev->chain_device, funcName);
+                if (gdpa_value != NULL) ldev->loader_dispatch.ext_dispatch[idx] = (PFN_vkDevExt)gdpa_value;
+                ldev = ldev->next;
+            }
+        }
+    }
+}
+
+// Find all dev extension in the hash table  and initialize the dispatch table
+// for dev  for each of those extension entrypoints found in hash table.
+void loader_init_dispatch_dev_ext(struct loader_instance *inst, struct loader_device *dev) {
+    for (uint32_t i = 0; i < MAX_NUM_UNKNOWN_EXTS; i++) {
+        if (inst->dev_ext_disp_functions[i] != NULL)
+            loader_init_dispatch_dev_ext_entry(inst, dev, i, inst->dev_ext_disp_functions[i]);
+    }
+}
+
+bool loader_check_icds_for_dev_ext_address(struct loader_instance *inst, const char *funcName) {
+    struct loader_icd_term *icd_term;
+    icd_term = inst->icd_terms;
+    while (NULL != icd_term) {
+        if (icd_term->scanned_icd->GetInstanceProcAddr(icd_term->instance, funcName))
+            // this icd supports funcName
+            return true;
+        icd_term = icd_term->next;
+    }
+
+    return false;
+}
+
+bool loader_check_layer_list_for_dev_ext_address(const struct loader_layer_list *const layers, const char *funcName) {
+    // Iterate over the layers.
+    for (uint32_t layer = 0; layer < layers->count; ++layer) {
+        // Iterate over the extensions.
+        const struct loader_device_extension_list *const extensions = &(layers->list[layer].device_extension_list);
+        for (uint32_t extension = 0; extension < extensions->count; ++extension) {
+            // Iterate over the entry points.
+            const struct loader_dev_ext_props *const property = &(extensions->list[extension]);
+            for (uint32_t entry = 0; entry < property->entrypoint_count; ++entry) {
+                if (strcmp(property->entrypoints[entry], funcName) == 0) {
+                    return true;
+                }
+            }
+        }
+    }
+
+    return false;
+}
+
+void loader_free_dev_ext_table(struct loader_instance *inst) {
+    for (uint32_t i = 0; i < inst->dev_ext_disp_function_count; i++) {
+        loader_instance_heap_free(inst, inst->dev_ext_disp_functions[i]);
+    }
+    memset(inst->dev_ext_disp_functions, 0, sizeof(inst->dev_ext_disp_functions));
+}
+
+/*
+ * This function returns generic trampoline code address for unknown entry points.
+ * Presumably, these unknown entry points (as given by funcName) are device
+ * extension entrypoints.  An array is used to keep a list of unknown entry
+ * points and their mapping to the device extension dispatch table
+ * (struct loader_dev_ext_dispatch_table).
+ * \returns
+ * For a given entry point string (funcName), if an existing mapping is found
+ * the  * trampoline address for that mapping is returned. Otherwise, this unknown
```

Comment:
Looks like a clang format might have messed this comment up.  Lots of ' * ' in the middle of the comment